### PR TITLE
Enable read method again with the new add API

### LINKED ISSUE #1232: Add `Model.read` again to Ribasim Python
`Model.read` wasn't migrated when working on the add API.
Let's add it back again.

### PR DESCRIPTION
**Pull request by evetion**
Re-introduce database context (originally in removed Network node) and split Node tables per Node. Fixes #1232

### CODE REVIEW

File: python/ribasim/tests/test_model.py [hunk 1]
```diff
@@ -155,7 +155,7 @@ def test_write_adds_fid_in_tables(basic, tmp_path):
     # for node an explicit index was provided
     nrow = len(model_orig.basin.node.df)
     assert model_orig.basin.node.df.index.name is None
-    assert model_orig.basin.node.df.index.equals(pd.Index(np.full(nrow, 0)))
+    # assert model_orig.basin.node.df.index.equals(pd.Index(np.full(nrow, 0)))
```
Comment (evetion) [inline python/ribasim/tests/test_model.py:158]: @Hofer-Julian I needed to reset the index after filtering the table to make write work, but that fails this test. Shall we remove this one?

Comment (Hofer-Julian) [reply python/ribasim/tests/test_model.py:158]: Yes, let's remove it.

**Review by Hofer-Julian — COMMENTED** [review]
Can you please enable all tests with reason "Needs Model read implementation"?

File: python/ribasim/ribasim/geometry/edge.py [hunk 2]
```diff
@@ -42,10 +42,6 @@ class Config:
 class EdgeTable(SpatialTableModel[EdgeSchema]):
     """Defines the connections between nodes."""
 
-    def __init__(self, **kwargs):
-        kwargs.setdefault("df", DataFrame[EdgeSchema]())
-        super().__init__(**kwargs)
```
Comment (Hofer-Julian) [inline python/ribasim/ribasim/geometry/edge.py:47]: This is needed.

Ribasim Core always requires an Edge table, even if empty.
Without this, no Edge table is added if you don't have edges

Comment (evetion) [reply python/ribasim/ribasim/geometry/edge.py:47]: Yep, but this interferes with the default Pydantic initialization, I fixed it in another way.

**Review by Hofer-Julian — APPROVED** [review]
After mypy is happy, this is ready to go from my side.
Thanks!

**Pull request merged**